### PR TITLE
colmap 3.8 (new formula)

### PR DESCRIPTION
**Pull request by njwhite**
- [x ] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x ] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x ] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/colmap.rb [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  depends_on "ceres-solver"
+  depends_on "cgal"
+  depends_on "eigen"
+  depends_on "flann"
+  depends_on "freeimage"
+  depends_on "gflags"
+  depends_on "glew"
+  depends_on "glog"
+  depends_on "lz4"
+  uses_from_macos "sqlite"
```
Comment (rgov) [inline Formula/colmap.rb:20]: [My PR](https://github.com/Homebrew/homebrew-core/pull/129705/files) also included `metis`, `qt@5`, and `suite-sparse`. (I didn't have `lz4`.) What is the consequence of omitting these, does this mean that the Qt GUI is not included?

I referred to [this document](https://colmap.github.io/install.html#id2) for the dependencies list.

File: Formula/colmap.rb [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+  def install
+    system "cmake", "-B", "build", ".", "-GNinja", *std_cmake_args
+    cd "build" do
+      system "ninja", "install"
+    end
```
Comment (rgov) [inline Formula/colmap.rb:26]: Can you use `cmake --build` instead? This seems to be preferred Homebrew style.

```ruby
    system "cmake", "--build", "build"
    system "cmake", "--install", "build"
```

File: Formula/colmap.rb [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+  depends_on "flann"
+  depends_on "freeimage"
+  depends_on "gflags"
+  depends_on "glew"
+  depends_on "glog"
+  depends_on "lz4"
+  uses_from_macos "sqlite"
+
+  def install
+    system "cmake", "-B", "build", ".", "-GNinja", *std_cmake_args
```
Comment (rgov) [inline Formula/colmap.rb:23]: Is it necessary to force the use of the Ninja generator? Maybe this is the secret to getting it to build on Linux?

Comment (chenrui333) [reply Formula/colmap.rb:23]: what is the upstream preference?

File: Formula/colmap.rb (hunk 2 above)
Comment (chenrui333) [reply Formula/colmap.rb:26]: yeah, definitely, can you make that change?

File: Formula/colmap.rb (hunk 1 above)
Comment (chenrui333) [reply Formula/colmap.rb:20]: I am not sure i understand your question.

Comment (rgov) [reply Formula/colmap.rb:20]: How did you derive the list of dependencies? The GUI relies on Qt, doesn't it?

**Comment by chenrui333** [comment]
#141878

**Pull request closed without merging**